include/zephyr/drivers/stepper.h

```diff
@@ -0,0 +1,241 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEPPER_H
+#define ZEPHYR_INCLUDE_STEPPER_H
+
+/** \addtogroup stepper
+ *  @{
+ */
+
+/**
+ * @file
+ * @brief Public API for Stepper Motor Controller
+ *
+ */
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Stepper Motor micro step resolution options
+ */
+enum micro_step_resolution {
+	STEPPER_FULL_STEP,
+	STEPPER_MICRO_STEP_2,
+	STEPPER_MICRO_STEP_4,
+	STEPPER_MICRO_STEP_8,
+	STEPPER_MICRO_STEP_16,
+	STEPPER_MICRO_STEP_32,
+	STEPPER_MICRO_STEP_64,
+	STEPPER_MICRO_STEP_128,
+	STEPPER_MICRO_STEP_256,
+};
+
+/**
+ * @brief Attibutes to set/get as per the functionality offered by the stepper controller
+ */
+enum stepper_attribute {
+	/** read/write actual a.k.a reference position in the stepper driver */
+	STEPPER_ACTUAL_POSITION,
+	/** read/write target a.k.a absolute position in the stepper driver */
+	STEPPER_TARGET_POSITION,
```

Comment:
Out of curiosity, will this be given as an accumulated number of steps or not? And how will the sign of this attribute be interpreted?

i.e.: if the motor made 2 full clockwise rotations and has a resolution of 200 steps, will this return 0, 400 or -400?

Reply:
With most stepper controllers if the motor moves in the positive direction, then the actual position step counter increases. If it moves in the negative direction, then the counter decreases. Most stepper drivers treat this as a signed value. However, some applications (less common) can ignore the sign and use  overflow/underflow for intentional wrap-around.

I think sticking to simply positive and negative steps is best. Using terminology like clockwise and counterclockwise is not as broad (and confusing for something like a stepper-driven linear actuator) and relies on how the motor is wired/connected.

Reply:
Sure! I tend to also use absolute steps, but saw some wanderers that prefer to wrap around, so it seems that the driver should enforce one or another to avoid such different interpretations